contracts/cosmwasm-vm/cw-xcall/src/handle_call_message.rs

```diff
@@ -27,6 +27,37 @@ impl<'a> CwCallService<'a> {
         }
     }
 
+    pub fn handle_reply(
+        &self,
+        deps: DepsMut,
+        rollback: CallRequest,
+        request: CSMessageRequest,
+    ) -> Result<Event, ContractError> {
+        if !(rollback.from() == request.to() && rollback.to().nid() == request.from().nid()) {
```

Comment:
isnt from here a NID while to Is a NetworkAddress?
Should be to .nid() i think. 
This would be caught in unit testing